front/production_checks/checks/managed_data_sources_gc.ts

```diff
@@ -0,0 +1,186 @@
+  checkName,
+  reportSuccess,
+  reportFailure
+) => {
+  const core_sequelize = new Sequelize(CORE_DATABASE_URI_RO as string, {
+    logging: false,
+  });
+  const front_sequelize = new Sequelize(FRONT_DATABASE_URI_RO as string, {
+    logging: false,
+  });
+  const connectorsSequelize = new Sequelize(
+    CONNECTORS_DATABASE_URI_RO as string,
+    {
+      logging: false,
+    }
+  );
+
+  const managedDsData = await front_sequelize.query(
+    'SELECT id, "connectorId", "connectorProvider", "dustAPIProjectId"\
+     FROM data_sources WHERE "connectorId" IS NOT NULL',
+    { type: QueryTypes.SELECT }
+  );
+  const managedDs = managedDsData as {
+    id: number;
+    connectorId: number;
+    connectorProvider: string;
+    dustAPIProjectId: string;
+  }[];
+  for (const ds of managedDs) {
+    const coreDsData = await core_sequelize.query(
+      `SELECT id FROM data_sources WHERE "project" = :dustAPIProjectId`,
+      {
+        replacements: {
+          dustAPIProjectId: ds.dustAPIProjectId,
+        },
+        type: QueryTypes.SELECT,
+      }
+    );
+    const coreDs = coreDsData as { id: number }[];
+    if (coreDs.length === 0) {
+      await reportFailure(
+        { frontDataSourceId: ds.id },
+        `No core data source found for front data source`
+      );
+      continue;
+    }
+    const coreDocumentsData = await core_sequelize.query(
+      `SELECT id, document_id, parents FROM data_sources_documents WHERE "data_source" = :coreDsId AND status = 'latest'`,
+      {
+        replacements: {
+          coreDsId: coreDs[0].id,
+        },
+        type: QueryTypes.SELECT,
+      }
+    );
+
+    const coreDocuments = coreDocumentsData as CoreDSDocument[];
+    const checkFiles =
+      CHECK_FILES_BY_TYPE[ds.connectorProvider as ConnectorProvider];
+    if (checkFiles) {
+      const result = await checkFiles(
+        ds.connectorId,
+        coreDocuments,
+        connectorsSequelize
+      );
+      if (result.isErr()) {
+        await reportFailure(
+          {
+            frontDataSourceId: ds.id,
+            coreDataSourceId: coreDs[0].id,
+            conncetorId: ds.connectorId,
+            connectorProvider: ds.connectorProvider,
+            errorPayload: result.error,
+          },
+          `Files from managed Data Sources not garbage collected.`
+        );
+      } else {
+        await reportSuccess({ frontDataSourceId: ds.id });
+      }
+    }
+  }
+};
+
+export const CHECK_FILES_BY_TYPE: Record<
+  ConnectorProvider,
+  DocumentsChecker | undefined
+> = {
+  slack: async (
+    connectorId: ModelId,
+    documents: CoreDSDocument[],
+    connectorsSequelize: Sequelize
+  ): Promise<Result<void, unknown>> => {
+    const missingChannels: string[] = [];
+    const coreChannel = new Set(documents.map((d) => d.parents[0]));
+    for (const channelId of coreChannel) {
+      const selectedChannel: { id: number; permission: string }[] =
+        await connectorsSequelize.query(
```

Comment:
How is this independent from the permission flag?

Reply:
Will handle Slack in a different PR. This was a mistake from my side.